### ISSUE TITLE
CDDA Modpack Structure Update - Moving to Bright Nights 

### ISSUE THREAD

**Issue by Kenan2000** [Kenan2000/CDDA-Structured-Kenan-Modpack#245]
@adamkad1, @CalKerethi, @jackledead, @Malkeus, @Saicchi, @waveyl, @Theawesomeboophis, @linonetwo, @CKRainbow, @Entity-y, @meelock, @Regularitee, @DeadLeaves, @Roet718, @BrettDong, @dranitski, @ProfoundDarkness, @TheRagingAnimist, @usagirei, @Malkeus, @321nick, @Domonor, @Phenomphear, @Szara-ManOfHonor, @Chaosvolt, @SzQ1, @AxemaVales

To all of your attention, contributors, players, modders, I really appreciate your support, constant help with fixes, tweaks, your kind words about my work, obviously, thanks to modders who allow me to include their mods in my structured Modpacks, however, I've made a decision that I don't want to continue updating this many mods with constant heavy loads of work for CDDA. Maybe you might ask me, why I've made such a decision? 

**Thing is, developers along with key contributors often show to have an unexplainable contempt towards 3rd party work aka**

**"Don't play it, never play it at all, it is of low quality, likely it is broken or barely works, causes crashes, errors" instead of**

**"We don't hold responsibility for quality or performance of 3rd party work, we are not associated with it at all, if you consider it low quality or experience errors or crashes, we could advise you how to fix it if it's a mod for our game if you want to fix it yourself, or if you can't fix it, try to write to author so he can fix it" honestly**

It's just they are not even involved with this work, if anything, they should be neutral towards it, because it's an attempt to make mods for their own game, because modders apparently like their game, so we make or maintain mods for it, showing contempt in a rude way towards 3rd party work modders, advising players against playing this work, what's the point of doing anything that requires plenty of free time, energy along with work where developers literally shun upon such creators or maintainers for unknown reasons? Why even developers allow such attitude if it has nothing to do with their own core game at all? 

**I will only try to update mods for CDDA that are made by @AxemaVales, one of my modding teachers, who I owe because he taught me a lot, without his help, I would not have learned how to tackle complex modding, I would have never created BL9 at all, I owe him along with some other people who show me huge support, such as @mng-cataclysm along with @eirenexthedragon, thus, I will try, with God's help, to continue updating all mods by Axema in my new High-Maintenance-Huge-Mods folder along with MNG's mods along with Eire's mods along with my own mods, projects, such as BL9 including some other small mods that I work on or try to maintain**

**I've made a decision to move to Bright Nights to continue updating my structured Bright Nights Modpack with actual folder structure, mods maintenance priority along with other important changes, you can find it using this https://github.com/Kenan2000/BrightNights-Structured-Kenan-Modpack, here**

**All mods by @chaosvolt for CDDA you can find in his repositories, however, I will try to make sure, again, with God's help, to continue updating his mods for Bright Nights in a friendlier environment where there is no strange hostility coming from developers towards 3rd party work or projects in a very convincing unpolite way to not use 3rd party work at all instead of advising to fix it themselves or make an issue so author can fix it**

**I don't want to force or convince anyone to move to Bright Nights, neither players, nor modders, you are here to judge or to choose what to play, if you prefer realism along with tons of micromanagement systems to take into consideration, some may find them tedious, some may find them fun or even unique, interesting, unlike any other game you've played, a thing is, CDDA strays away from roguelike genre using it's deep realism mechanics along with micromanagement complexity**

**Bright Nights is much closer to what a roguelike is based on it's gameplay, it retains a lot of CDDA's complexity, most of it, though, instead of concentrating on realism or micromanagement, it concentrates most on gameplay along with more global scale management, new features or systems that provide more variety vs micromanagement, some may find it easier considering that it tries to simplify overly complicated micromanagement, but in favor it concentrates on other important features which make gameplay just as varied or complex as CDDA for your knowledge**

If you want to discuss this or have any questions, feel free to ask or write it as well

**Comment by SzQ1** [comment]
Tldr: Kenan stops updating DDA mods, will only update mods for BN.

**Comment by MikuMiddo** [comment]
Unbengable

**Comment by chaosvolt** [comment]
When it comes to compilations and other repositories, the people raising these objections hypothetically have a point in that managing multiple competing sources for any given mod or other resource can lead to some serious technical issues. I've voiced some complaints in the past for outdated features being preserved and mod desync happening, and a good percentage of bug reports I get tend to be false positives caused by people using an outdated version of my mod, or a fix filtering into a compilation but not the source repo.

However, this isn't a black mark on Kenan's competence as a compilation, nor something unique to mod compliations. In the old days, Remyroy's launcher was the source of 90% of bug reports I got, and the answer was typically always "update your stuff manually because by design the launcher has no provision for updating third-party mods." It's just an unavoidable risk that comes with ANYTHING that can allow for an inexperienced user to grab the wrong version of a mod for the version of a game they're using.

Hell, these days a decent chunk of my bug reports instead come from users who grabbed the master branch of my mods with stable DDA, vice-versa using a stable release of a mod with experimental DDA, or getting the DDA and BN versions of my mods mixed up.

But the point raised in the OP here highlights this: **The way these concerns have been raised is completely unacceptable, and the DDA devs have consistently sought to burn bridges with any part of their community not under their direct authority.**

With that in mind, it's probably for the best to not make things too much harder for yourself in terms of effort. Maintaining mods is a pain in the ass for just one person, so shifting focus to more manageable projects is a good idea, especially if achieving that lets you drop a particularly toxic community that actively makes your work harder.

As for my mods, I'll continue keeping both BN and DDA versions up to date and functional so long as it's convenient for me to do so. For now it's tolerable due to the way I have my repositories set up, in particular that's why I have BN and DDA versions of any given mod as part of the same repo. Despite the occasional false-positive bug report it causes, I probably would've dropped support for most of my DDA mods by now if I didn't do things that way.

**Comment by chaosvolt** [comment]
Additionally, for those wondering, much of the current stance on mod compilations can be seen here: https://github.com/CleverRaven/Cataclysm-DDA/wiki/FAQ-from-Discord#Mods_and_Mod_Packs

Their phrasing there is less abrasive than the way they usually phrase things elsewhere such as on the development server, but you can see hints of it in sections such as this:

> E.g. Kenan’s pack advertises having an ‘up to date’ version of Blaze Mod. But the stuff in that one that isn’t in the new, in-repo, Blaze Industries version of the mod was cut for a reason — primarily stability, but also for using hacky workarounds, or trying to implement stuff that doesn’t even work.

Directly naming names, especially when other mod compiliations will do the same thing, serves no purpose. And if you have specific examples of a problem with a mod compliation, then it would be more productive to actually point them out to the repository maintainers and ask them to fix things.

Hell, @SzQ1 has been doing exactly that with this exact example given, working on his own updated version of Blazemod to get all of its original features working in a form that's more up to the standard of modern Cataclysm mods.

Instead, the writer of this particular section (Mom-Bun in this case) has decided to uphold this specific problem as an example of poor quality, and probably stands to benefit from any perceived problems with it staying unfixed to continue serving as an example, and probably would continue to do so even if the old version of blazemod were completely removed in favor of SzQ's version.

**Comment by Theawesomeboophis** [comment]
Will you still accept PRs? Or is this the end? If so, I really appreciate all the work that goes into this modpack and without your work, Cdda would be a lot less enjoyable for a lot of people, I thank you sincerely for maintaining these mods and for otopack, you have improved my experience with the game.

**Comment by chaosvolt** [comment]
For anything still being actively maintained, it'd be better to keep an eye on the relevant repos.

**Comment by Theawesomeboophis** [comment]
But what about mods no longer maintained, such as the aforementioned og Blazemod?

**Comment by SzQ1** [comment]
What is actual problem with blazemod in DDA @Theawesomeboophis ?

**Comment by Theawesomeboophis** [comment]
I'm not sure, it seems to work fine to me, it's not particularly gamebreaking or unstable as far as I know, I've been playing the game with blazemod for a while and it works pretty well.

**Comment by Kenan2000** [comment]
> Will you still accept PRs? Or is this the end? If so, I really appreciate all the work that goes into this modpack and without your work, Cdda would be a lot less enjoyable for a lot of people, I thank you sincerely for maintaining these mods and for otopack, you have improved my experience with the game.

@Theawesomeboophis I really appreciate your kind words, Theawesomeboophis, no, as I said, I've only reduced my amount of workload for CDDA from 100% to 10%, concentrating only on most important projects by people whose mods don't often get updated or they find it hard to keep up with CDDA updates along with my own projects

I will make sure to try to maintain mods that Axema has made, that MNG has made, that Eire has made, of course, hopefully, I will make sure to try to not only maintain, but also update my own projects, these people that I've mentioned are very kind, also constantly showing support, I feel that I cannot just abandon their works, especially work that Axema has made, because he really taught me a lot of things for advanced modding in Cataclysm, without his work I would have never created BL9, would have had much more problems fixing errors because of lack of JSON knowledge, so it is my obligation to make sure to continue to try to maintain these mods 

@chaosvolt You are partially right about relevant repos statement, however, to avoid misunderstanding, this post doesn't mean that I want to abandon working on all CDDA mods that I've been trying to maintain, I've cut amount of workload for me, including your mods, having notified you, thanks to your quality maintenance of your mods along with their popularity, I don't need to include it in CDDA Modpack, it is nice for both of us because, as you see, I prefer to concentrate on most important projects of my CDDA Modpack that are made by people that cannot really keep up with updating them often along with my own projects 

I've also restructured my CDDA Modpack, so what is in High-Maintenance-Huge-Mods is priority mods for me to update if they break or crash or throw errors, however, your statement fits very nicely for Archived-Mods, which is left for whoever is interested to pick then update, in which case, if any author or new maintainer of mods in Archived-Mods is interested in maintaining a mod from it, please, make sure to notify me so I can delete it from this folder, it is here only for preserving mods for whoever wants to continue updating them for latest experimental versions as well

**Comment by Theawesomeboophis** [comment]
That's a shame, It was a pleasure working with all of you.

**Comment by Kenan2000** [comment]
> That's a shame, It was a pleasure working with all of you.

@Theawesomeboophis Hey, sorry for misunderstanding, mate I mean, yes, I will accept PRs, only for High-Maintenance-Huge-Mods though, but not for Archived-Mods, because if you want to update them, I will then just delete them from my repository so they can be attributed to actual person who maintains them because I have no interest in fixing or looking into them, that's why I've put them in this folder to specify that it's of no interest to me in updating them as well

**Comment by Theawesomeboophis** [comment]
Oh, I thought you meant no as in you weren't accepting PRs anymore, I'm glad this is not the case.

**Comment by Malkeus** [comment]
I'll admit, I didn't read this before submitting the PR for dinoMod extended... I'll just make a repo, this interface must be new, makes things easy

**Comment by Kenan2000** [comment]
Hmm, if anyone wants to make any new mods, yeah, I prefer if you keep them on your own repository, it's already plenty of work for me to maintain current, small amount of mods for CDDA, which is why I don't expand it, I try to maintain what I personally, knowing original authors, consider the most important third party mods to maintain as well

**Comment by Malkeus** [comment]
It's not new, I got it from gmc, but goat hasn't updated that in months. Piles of prs waiting for attention.
I think trying to keep pace with Experimental updates is futile, especially if you're looking at large amounts of mods. I'm going to try a new meta in my repo. It will contain a link to a specific version of cataclysm (the one I'm currently playing, most likely) and the mods within will be only those that work without any load errors (super strict) on that version.
The goal is for a newbie to cataclysm to be able to grab a full mod pack and get linked to the version of cataclysm it works on all in one spot. 
Its essentially just sharing what I already do every few months with the community. I usually end up losing my work to various stuff, this way it will go somewhere useful. I just updated earlier this week and I've got my old files still this time around... Watch the skies for me.

**Comment by Kenan2000** [comment]
We don't even try to keep pace with experimental updates, we strive to be a few steps back from them at least, CDDA will always be one step ahead of third party mods. It is important to update mods at our own pace, we do this for free, spending time, energy, we have a right to update mods at our own pace, update mods with pace that we choose ourselves, try to maintain smaller amount of mods if bigger workload is too much energy or bigger amount of mods if we are confident that we can try to be at least a few steps back from latest experimental version. 

I have a made a decision to reduce amount of mods that I maintain for CDDA because I've moved to Bright Nights mostly, I try to maintain this small amount of mods because I try to be at least a few steps back from latest experimental updates for mods, I value authors of these mods, they support me, some have taught me modding a lot. We set this pace ourselves, I can prioritize updates for mods updating some mods later, this is still work that I try to do that players appreciate, so our own pace is the most important thing, any pace that one keeps works, players appreciate any pace of work that we, you, I try to keep for third party mods hopefully as well
